observatory-dags/observatory/dags/telescopes/oapen_irus_uk.py

```diff
@@ -0,0 +1,459 @@
+                                                 max_instances, update)
+            if success:
+                logging.info(f'Creating or patching cloud function successful, response: {msg}')
+            else:
+                raise AirflowException(f'Creating or patching cloud function unsuccessful, error: {msg}')
+        else:
+            logging.info(f'Using existing cloud function, source code has not changed.')
+
+    def call_cloud_function(self, oapen_project_id: str, source_bucket: str, function_name: str, function_region: str):
+        """ Call the google cloud function that is inside the oapen project id
+
+        :param oapen_project_id: The oapen project id.
+        :param source_bucket: Storage bucket with the source code
+        :param function_name: Name of the google cloud function
+        :param function_region: Region where the google cloud function resides
+        :return: None.
+        """
+        function_url = f'https://{function_region}-{oapen_project_id}.cloudfunctions.net/{function_name}'
+        geoip_license_key = BaseHook.get_connection(AirflowConns.GEOIP_LICENSE_KEY).password
+
+        if self.release_date >= datetime(2020, 4, 1):
+            publisher_uuid = get_publisher_uuid(self.publisher_name)
+            airflow_conn = AirflowConns.OAPEN_IRUS_UK_API
+        else:
+            publisher_uuid = "NA"
+            airflow_conn = AirflowConns.OAPEN_IRUS_UK_LOGIN
+        username = BaseHook.get_connection(airflow_conn).login
+        password = BaseHook.get_connection(airflow_conn).password
+
+        success = call_cloud_function(function_url, self.release_date.strftime('%Y-%m'), username, password,
+                                      geoip_license_key, self.publisher_name, publisher_uuid, source_bucket,
+                                      self.blob_name)
+        if not success:
+            raise AirflowException('Cloud function unsuccessful')
+
+    def transfer(self, oapen_bucket: str):
+        """ Transfer blob from bucket inside oapen project to bucket in airflow project.
+
+        :param oapen_bucket: Name of the oapen bucket
+        :return: None.
+        """
+        success = copy_blob_from_cloud_storage(self.blob_name, oapen_bucket, self.download_bucket)
+        if not success:
+            raise AirflowException('Transfer blob unsuccessful')
+
+    def download_transform(self):
+        """ Download blob with access stats to a local file.
+
+        :return: None.
+        """
+        success = download_blob_from_cloud_storage(self.download_bucket, self.blob_name, self.transform_path)
+        if not success:
+            raise AirflowException('Download blob unsuccessful')
+
+
+class OapenIrusUkTelescope(SnapshotTelescope):
+    DAG_ID_PREFIX = 'oapen_irus_uk'
+    ORG_MAPPING = {'ucl_press': quote("UCL Press"),
+                   'anu_press': quote("ANU Press"),
+                   'wits_university_press': quote("Wits University Press")}
```

Comment:
As with the other telescopes, it would be good to store these values as part of the `extra` field of the Telescope via the Observatory API and pass the `publisher_id` variable explicitly to the OapenIrusUkTelescope constructor.